### PR TITLE
CORDA-716 Move non-API things to internal

### PR DESCRIPTION
**Pull request by andr3ej**

### CODE REVIEW

File: client/rpc/src/test/kotlin/net/corda/client/rpc/Measure.kt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package net.corda.testing
+package net.corda.client.rpc
```
Comment (andr3ej) [inline client/rpc/src/test/kotlin/net/corda/client/rpc/Measure.kt]: moved to the only place it's used

File: node-api/src/test/kotlin/net/corda/nodeapi/Eventually.kt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package net.corda.testing
+package net.corda.nodeapi
```
Comment (andr3ej) [inline node-api/src/test/kotlin/net/corda/nodeapi/Eventually.kt]: it's not common code if only 1 thing uses it

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNetworkMapCache.kt [hunk 3]
```diff
@@ -1,39 +0,0 @@
...
-    override val changed: Observable<NetworkMapCache.MapChange> = PublishSubject.create<NetworkMapCache.MapChange>()
-    override val nodeReady: CordaFuture<Void?> get() = doneFuture(null)
-
-    init {
-        val mockNodeA = NodeInfo(listOf(BANK_C_ADDR), listOf(BANK_C), 1, serial = 1L)
-        val mockNodeB = NodeInfo(listOf(BANK_D_ADDR), listOf(BANK_D), 1, serial = 1L)
-        addNode(mockNodeA)
-        addNode(mockNodeB)
-    }
-}
```
Comment (andr3ej) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/MockNetworkMapCache.kt]: all this when only nodeReady was needed!

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt [hunk 4]
```diff
@@ -115,8 +115,6 @@ open class MockServices private constructor(
     constructor(identityService: IdentityServiceInternal, initialIdentityName: CordaX500Name, vararg keys: KeyPair) : this(emptyList(), identityService, initialIdentityName, *keys)
     constructor(identityService: IdentityServiceInternal, initialIdentityName: CordaX500Name) : this(identityService, initialIdentityName, generateKeyPair())
 
-    val key: KeyPair get() = keys.first()
```
Comment (andr3ej) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt]: i guess a test may want to get the autogenerated key, but it's easy and less magic for the test to generate one itself and pass it in, especially now we have TestIdentity

File: testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt [hunk 5]
```diff
@@ -132,7 +130,7 @@ open class MockServices private constructor(
     override val clock: Clock get() = Clock.systemUTC()
     override val myInfo: NodeInfo
         get() {
-            val identity = getTestPartyAndCertificate(initialIdentityName, key.public)
+            val identity = getTestPartyAndCertificate(initialIdentityName, keys.first().public)
```
Comment (andr3ej) [inline testing/node-driver/src/main/kotlin/net/corda/testing/node/MockServices.kt]: this gets nicer in #2247

File: testing/test-utils/src/main/kotlin/net/corda/testing/CoreTestUtils.kt [hunk 6]
```diff
@@ -126,20 +120,6 @@ class TestIdentity @JvmOverloads constructor(val name: CordaX500Name, entropy: L
...
-} else {
-    loggerFor<T>().info("Ignoring Kryo specific test, reason: $reason")
-}
-
-@Suppress("unused")
-inline fun <reified T : Any> T.amqpSpecific(reason: String, function: () -> Unit) = if (AMQP_ENABLED) {
-    function()
-} else {
-    loggerFor<T>().info("Ignoring AMQP specific test, reason: $reason")
-}
```
Comment (andr3ej) [inline testing/test-utils/src/main/kotlin/net/corda/testing/CoreTestUtils.kt]: these don't look like api, moved to InternalTestUtils

File: testing/test-utils/src/main/kotlin/net/corda/testing/CoreTestUtils.kt [hunk 7]
```diff
@@ -156,25 +136,3 @@ fun NodeInfo.singleIdentityAndCert(): PartyAndCertificate = legalIdentitiesAndCe
...
- * it won't work unless you mock a (trivial) abstract implementation of that interface instead.
- */
-fun <T> rigorousMock(clazz: Class<T>): T = mock(clazz) {
-    if (Modifier.isAbstract(it.method.modifiers)) {
-        // Use ThrowsException to hack the stack trace, and lazily so we can customise the message:
-        ThrowsException(UndefinedMockBehaviorException("Please specify what should happen when '${it.method}' is called, or don't call it. Args: ${Arrays.toString(it.arguments)}")).answer(it)
-    } else {
-        it.callRealMethod()
-    }
-}
```
Comment (andr3ej) [inline testing/test-utils/src/main/kotlin/net/corda/testing/CoreTestUtils.kt]: not api, moved to InternalTestUtils

File: testing/test-utils/src/main/kotlin/net/corda/testing/TestDependencyInjectionBase.kt [hunk 8]
```diff
@@ -1,10 +0,0 @@
-package net.corda.testing
-
-import org.junit.Rule
-
-@Deprecated("Instead of extending this class, use SerializationEnvironmentRule in the same way.")
-abstract class TestDependencyInjectionBase {
-    @Rule
-    @JvmField
-    val testSerialization = SerializationEnvironmentRule()
-}
```
Comment (andr3ej) [inline testing/test-utils/src/main/kotlin/net/corda/testing/TestDependencyInjectionBase.kt]: do we really have to keep this as a deprecated thing? the code change is trivial

File: testing/test-utils/src/main/kotlin/net/corda/testing/TestTimestamp.kt [hunk 9]
```diff
@@ -1,20 +0,0 @@
...
-class TestTimestamp {
-    companion object {
-        val timestamp: String = {
-            val tz = TimeZone.getTimeZone("UTC")
-            val df = SimpleDateFormat("yyyyMMddHHmmss")
-            df.timeZone = tz
-            df.format(Date())
-        }()
-    }
-}
```
Comment (andr3ej) [inline testing/test-utils/src/main/kotlin/net/corda/testing/TestTimestamp.kt]: unused

File: testing/test-utils/src/main/kotlin/net/corda/testing/internal/vault/DummyDealContract.kt [hunk 10]
```diff
@@ -1,4 +1,4 @@
-package net.corda.testing.contracts
+package net.corda.testing.internal.vault
```
Comment (andr3ej) [inline testing/test-utils/src/main/kotlin/net/corda/testing/internal/vault/DummyDealContract.kt]: this stuff is specific to VaultFiller and the handful of tests that use it. i don't think we should be encouraging users to use VaultFiller in its current state

File: testing/test-utils/src/main/kotlin/net/corda/testing/internal/vault/VaultFiller.kt [hunk 11]
```diff
@@ -1,6 +1,4 @@
-@file:JvmName("VaultFiller")
```
Comment (andr3ej) [inline testing/test-utils/src/main/kotlin/net/corda/testing/internal/vault/VaultFiller.kt]: it's a class now

File: node/src/test/java/net/corda/node/services/vault/VaultQueryJavaTests.java [hunk 12]
```diff
@@ -24,8 +24,8 @@
 import net.corda.nodeapi.internal.persistence.DatabaseTransaction;
 import net.corda.testing.SerializationEnvironmentRule;
 import net.corda.testing.TestIdentity;
-import net.corda.testing.contracts.DummyLinearContract;
-import net.corda.testing.contracts.VaultFiller;
+import net.corda.testing.internal.vault.DummyLinearContract;
```
Comment (roger3cev) [inline node/src/test/java/net/corda/node/services/vault/VaultQueryJavaTests.java]: why internal?

File: node/src/test/kotlin/net/corda/node/services/vault/VaultQueryTests.kt [hunk 13]
```diff
@@ -26,11 +26,14 @@ import net.corda.node.internal.configureDatabase
 import net.corda.nodeapi.internal.persistence.CordaPersistence
 import net.corda.nodeapi.internal.persistence.DatabaseConfig
 import net.corda.testing.*
-import net.corda.testing.contracts.*
+import net.corda.testing.internal.rigorousMock
+import net.corda.testing.internal.vault.DUMMY_LINEAR_CONTRACT_PROGRAM_ID
```
Comment (roger3cev) [inline node/src/test/kotlin/net/corda/node/services/vault/VaultQueryTests.kt]: why internal?

File: node/src/test/kotlin/net/corda/node/services/vault/VaultQueryTests.kt [hunk 14]
```diff
@@ -26,11 +26,14 @@ import net.corda.node.internal.configureDatabase
...
+import net.corda.testing.internal.vault.DummyLinearContract
```
Comment (roger3cev) [inline node/src/test/kotlin/net/corda/node/services/vault/VaultQueryTests.kt]: This bits and bobs can be used for cordapp tests, i think?

File: node/src/test/java/net/corda/node/services/vault/VaultQueryJavaTests.java (hunk 12 above)
Comment (andr3ej) [reply node/src/test/java/net/corda/node/services/vault/VaultQueryJavaTests.java]: iirc this is VaultFiller-specific stuff. the non VaultFiller specific stuff is still non-internal

Comment (andr3ej) [reply node/src/test/java/net/corda/node/services/vault/VaultQueryJavaTests.java]: not 100% sure so I won't merge just yet

File: node/src/test/kotlin/net/corda/node/services/vault/VaultQueryTests.kt (hunk 14 above)
Comment (andr3ej) [reply node/src/test/kotlin/net/corda/node/services/vault/VaultQueryTests.kt]: this particular one is only used in vault tests

**Pull request merged**